### ISSUE TITLE
void_function_in_ternary incorrectly triggered with if and switch expressions

### ISSUE THREAD

**Issue by tristanSIF** [realm/SwiftLint#5611]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

```swift
func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
    switch indexPath.section {
    // Void Function in Ternary Violation: Using ternary to call Void functions should be avoided (void_function_in_ternary)
    case 0: isEditing ? CGSize(width: 150, height: 20) : CGSize(width: 100, height: 20) 
    default: .zero
    }
}

func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
    if indexPath.section == 0 {
        // Void Function in Ternary Violation: Using ternary to call Void functions should be avoided (void_function_in_ternary)
        isEditing ? CGSize(width: 150, height: 20) : CGSize(width: 100, height: 20)
    } else {
        .zero
    }
}

### Environment

* SwiftLint version: 0.55.1
* Installation method used:  Homebrew

**Comment by mathieulb** [comment]
I second this, I have the problem with a function (x: MyEnum) -> String when using it in both sides of a ternary in a case. When I use the same function on only one side, the warning is not triggered. E.g.:

```
return switch {
  case 1: b ? f(.hello) : f(.world) // spurious warning
  case 2: b ? f(.hello) : "world" // OK
  case 3: b ? "hello" : f(.world) // OK
}
```

**Comment by vadimbelyaev** [comment]
Just encountered it with the latest SwiftLint version to date (0.59.1).
Looks like this issue and the linked PR are abandoned.

Here's my minimal example that triggers a false positive:
```swift
struct MyStruct {
    let someField: Int
}

func makeMyStruct() -> MyStruct {
    if true {
        true ? MyStruct(someField: 0) : MyStruct(someField: 1) // Void Function in Ternary Violation: Using ternary to call Void functions should be avoided (void_function_in_ternary)
    } else {
        true ? MyStruct(someField: 2) : MyStruct(someField: 3) // Void Function in Ternary Violation: Using ternary to call Void functions should be avoided (void_function_in_ternary)
    }
}
```

Just wondering of there's any chance to get it fixed.

**Comment by vadimbelyaev** [comment]
Another example I just discovered is with the string concatenation:
```swift
func makeString() -> String {
    "A"
}

func appendToString() {
    var str = ""
    str += true ? makeString() : makeString() // Void Function in Ternary Violation: Using ternary to call Void functions should be avoided (void_function_in_ternary)
}
```
